iothub/device/src/Serialization/DefaultPayloadConvention.cs

```diff
@@ -78,7 +78,17 @@ internal T GetObject<T>(Stream streamToConvert)
         [SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Method must be instance to benefit from s_settings.")]
         internal T GetObject<T>(string jsonObjectAsText)
         {
-            return JsonConvert.DeserializeObject<T>(jsonObjectAsText);
+            T deserialized;
+
+            try
+            {
+                deserialized = JsonConvert.DeserializeObject<T>(jsonObjectAsText);
+            }
+            catch (JsonReaderException)
+            {
+                deserialized = JsonConvert.DeserializeObject<T>(JsonConvert.SerializeObject(Encoding.UTF8.GetBytes(jsonObjectAsText)));
```

Comment:
Isn't this just a no-op? You are deserializing the thing you just serialized, right?

Reply:
The serialize-deserialize step ensures that we end up with type T as the final result. It's ugly, but if it were to be left out, there would be a compiler error saying that we're returning byte[] not type T (even though in the case of direct method payloads, they're the same).

Reply:
Gotcha. We actually hit this same problem in the e4k sdk repo, and this is what we settled on. Feel free to use what you already have, but please do leave a note in code explaining all this for future readers.

```suggestion
            // T should always be of type byte[] here, so this is basically a no-op.
            // However, .NET does not allow us to simply return "payload as T" or other basic casts.
                deserialized = (T)Convert.ChangeType(jsonObjectAsText, typeof(T), CultureInfo.InvariantCulture);
```